improver_tests/standardise/test_StandardiseMetadata.py

```diff
@@ -212,6 +212,46 @@ def test_air_temperature_status_flag_coord(self):
         self.assertArrayEqual(result.data, target.data)
         self.assertEqual(result.coords(), target.coords())
 
+    def test_air_temperature_status_flag_coord_without_realization(self):
+        """
+        Ensure we handle cubes which now include an 'air_temperature_status flag'
+        coord to signify points below surface altitude, where previously this
+        was denoted by NaN values in the data.
+
+        Tests this standardisation works if input is a deterministic forecast without
+        a realization coordinate.
+        """
+        cube = set_up_variable_cube(
+            np.full((3, 5, 5), fill_value=282, dtype=np.float32),
+            spatial_grid="latlon",
+            standard_grid_metadata="gl_det",
+            pressure=True,
+            height_levels=[100000.0, 97500.0, 95000.0],
+        )
+        # The target cube has 'NaN' values in its data to denote points below
+        # surface altitude.
+        result_no_sf = cube.copy()
+        result_no_sf.data[:, 0, ...] = np.nan
+        target = self.plugin.process(result_no_sf)
+
+        cube_with_flags = cube.copy()
+        flag_status = np.zeros((3, 5, 5), dtype=np.int8)
+        flag_status[:, 0, ...] = 1
```

Comment:
Just wondering if this would make more sense as shown below? My understanding would be that this would mean that the highest pressure level (which would have the lowest altitude) would all be set to the "below_surface_pressure" flag.
```suggestion
        flag_status[0, ...] = 1
```

Reply:
You're right that's how it should be, I also updated the indexing on line 234 so these match.